src/render/md.jl

```diff
@@ -30,14 +31,184 @@ function validate(mdstyle::Dict{Symbol, ByteString})
     end
 end
 
-function save(file::String, mime::MIME"text/md", doc::Metadata, mdstyle::Dict{Symbol, ByteString} = DEFAULT_MDSTYLE;
-                                                                                mathjax = false, include_internal = true)
+"Permalink Entry used when collection needed info."
+type PLEntry
+    objname::ByteString
+    idname::ByteString
+    descr::ByteString
+    relsourcepath::ByteString
+end
+
+"Permalink Main helper type."
+type PLMain
+    genidxfile::ByteString
+    mdoutfile::ByteString      # reference to the current proccessed output file needed for relative links.
+    idxrelpath::ByteString     # current relative path: mdoutfile to genidxfile
+    headerstyle::ByteString
+    modnamestyle::ByteString
+    data::Dict{ByteString, Dict{Symbol, Array{PLEntry,1}}}
+end
```

Comment:
Could we have more direct names for `PLEntry` and `PLMain`. Perhaps `Permalink` and `Index`?